martian/src/stage.rs

```diff
@@ -494,6 +494,23 @@ impl MartianRover {
             Ok(())
         }
     }
+
+    /// Creates a hard link (falls back to copy) to the given MartianFileType file
+    /// within the rover's file directory
+    ///
+    /// # Arguments
+    ///
+    /// * `src` - The path to the MartianFileType file
+    ///
+    /// # Returns
+    ///
+    /// * The path of the newly created link
+    pub fn hard_link_or_copy<M: MartianFileType>(&self, src: &M) -> Result<M, Error> {
```

Comment:
Do we actually need to narrow this to things that implement `MartianFileType` or would anything coercible to a path be more flexible?

Reply:
`src: impl AsRef<Path>` would be more flexible. But we need to:
1. Handle directories. Maybe we can recursively hard link?
2. Decide the return type. Either return a `PathBuf` or return `T: MartianMakePath` similar the `make_path` function.

`M: MartianFileType` essentially constraints the path to be a file and allows us to return the same type.

Reply:
Ah yeah I see, getting the same strong type in return position is quite nice.  Please leave it as-is.